### PR TITLE
Update to Bazel 1.1

### PR DESCRIPTION
**Pull request by aherrmann**
**depends on #1141 change base branch before merging**

This is based on and replaces https://github.com/tweag/rules_haskell/pull/1065.

- Updates nixpkgs to include Bazel 1.1.
- Updates the Stackage snapshot as in #1065 .
- Updates `rules_nixpkgs` to include `nixpkgs_python_configure`.
- Uses `nixpkgs_python_configure` to avoid issues with the Python toolchain as described in https://github.com/tweag/rules_haskell/pull/1065.
- Uses the nixpkgs platform constraint defined by rules_nixpkgs instead of defining its own.
    Defines an alias to be backwards compatible to users depending on that platform constraint.

Closes https://github.com/tweag/rules_haskell/pull/1065.

### CODE REVIEW

File: .netlify/build.sh [hunk 1]
```diff
@@ -23,8 +23,7 @@ mv WORKSPACE.tmp WORKSPACE
 # generation either.
 sed -i 's/vendored_node = "@nixpkgs_nodejs"/vendored_node = None/' WORKSPACE
 
-# XXX: see .bazelrc note [backward compatible options] for rational behind
-# the --incompatible_use_python_toolchains flag
-bazel build --incompatible_use_python_toolchains=false //docs:api_html
+# XXX: Remove --host_force_python=PY2 after updating to Starlark 0.4.0
```
Comment (mboes) [inline .netlify/build.sh]: What's Starlark 0.4? Is the language now versioned separately from Bazel?

File: WORKSPACE [hunk 2]
```diff
@@ -239,43 +232,43 @@ cc_library(
...
 )
 
-#nixpkgs_package(
-#    name = "python3",
-#    attribute_path = "python3",
-#    repository = "@nixpkgs",
-#)
+nixpkgs_package(
+    name = "python3",
+    attribute_path = "python3",
```
Comment (mboes) [inline WORKSPACE]: Nit: I believe this line is line is redundant.

File: .netlify/build.sh (hunk 1 above)
Comment (mboes) [reply .netlify/build.sh]: Changes below suggest you meant Stardoc.

File: WORKSPACE [hunk 3]
```diff
@@ -211,20 +200,24 @@ register_toolchains(
...
 
 nixpkgs_cc_configure(
     nix_file = "//nixpkgs:cc-toolchain.nix",
-    repository = "@nixpkgs",
+    nix_file_deps = ["//nixpkgs:default.nix"],
+    repository = "@nixpkgs_default",
+)
+
+nixpkgs_python_configure(
+    # XXX: Remove python2_attribute_path after updating to Starlark 0.4.0
```
Comment (mboes) [inline WORKSPACE]: Same as above: did you mean Stardoc?

File: haskell/repositories.bzl [hunk 4]
```diff
@@ -24,9 +24,9 @@ def rules_haskell_dependencies():
     if "io_tweag_rules_nixpkgs" not in excludes:
         http_archive(
             name = "io_tweag_rules_nixpkgs",
-            sha256 = "5a384daa57b49abf9f0b672852f1a66a3c52aecf9d4d2ac64f6de0fd307690c8",
-            strip_prefix = "rules_nixpkgs-0.5.2",
-            urls = ["https://github.com/tweag/rules_nixpkgs/archive/v0.5.2.tar.gz"],
+            sha256 = "b13cb651e972b3aa06bab75603923071d7647c9ee3f8acb707e3831aa606e911",
+            strip_prefix = "rules_nixpkgs-bdc802f8e716c66d50cc91f7ad22505edf13c48b",
```
Comment (mboes) [inline haskell/repositories.bzl]: Ideally we should cut a new release of upstream.

File: .netlify/build.sh (hunk 1 above)
Comment (aherrmann) [reply .netlify/build.sh]: Good catch. Yes, I meant stardoc.

**Pull request merged**